Pi+/Coxeter/Lists.agda

```diff
@@ -85,6 +85,21 @@ head+tail idp idp = idp
 start+end : {h1 h2 : List} -> {t1 t2 : List} -> (h1 == h2) -> (t1 == t2) -> (h1 ++ t1) == (h2 ++ t2)
 start+end idp idp = idp
 
+cut-last : {a1 a2 : ℕ} -> {l1 l2 : List} -> (l1 ++ [ a1 ] == l2 ++ [ a2 ]) -> (l1 == l2)
+cut-last {a1} {.a1} {nil} {nil} idp = idp
+cut-last {a1} {a2} {nil} {x :: nil} ()
+cut-last {a1} {a2} {nil} {x :: x₁ :: l2} ()
+cut-last {a1} {a2} {x :: nil} {nil} ()
+cut-last {a1} {a2} {x :: x₁ :: l1} {nil} ()
+cut-last {a1} {a2} {x :: l1} {x₁ :: l2} p = head+tail (cut-tail p) (cut-last (cut-head p))
+
+cut-prefix : {a1 a2 : ℕ} -> {l1 l2 : List} -> (l1 ++ [ a1 ] == l2 ++ [ a2 ]) -> (a1 == a2)
+cut-prefix {a1} {.a1} {nil} {nil} idp = idp
+cut-prefix {a1} {a2} {nil} {x :: nil} ()
+cut-prefix {a1} {a2} {nil} {x :: x₁ :: l2} ()
+cut-prefix {a1} {a2} {x :: nil} {nil} ()
+cut-prefix {a1} {a2} {x :: x₁ :: l1} {nil} ()
+cut-prefix {a1} {a2} {x :: l1} {x₁ :: l2} p = cut-prefix (cut-head p)
 
```

Comment:
These lemmas have better proofs using the path space of lists, but we can ignore that for now.

Reply:
You can change that, I will be happy to learn!